route/build/src/kwt_route.f90

```diff
@@ -408,216 +409,218 @@ SUBROUTINE QROUTE_RCH(IENS,JRCH,    & ! input: array indices
- integer(i4b), intent(in)                    :: ixOutlet ! index of the outlet reach
- INTEGER(I4B), INTENT(IN)                    :: JRCH    ! reach to process
- REAL(DP), INTENT(IN)                        :: T0,T1   ! start and end of the time step (seconds)
- INTEGER(I4B), INTENT(IN)                    :: LAKEFLAG ! >0 if processing lakes
- INTEGER(I4B), INTENT(IN), OPTIONAL          :: RSTEP   ! retrospective time step offset
- ! output variables
- integer(i4b), intent(out)                   :: ierr    ! error code
- character(*), intent(out)                   :: message ! error message
- ! (1) extract flow from upstream reaches and append to the non-routed flow in JRCH
- INTEGER(I4B)                                :: NUPS    ! number of upstream reaches
- REAL(DP),DIMENSION(:),POINTER,SAVE          :: Q_JRCH  ! flow in downstream reach JRCH
- REAL(DP),DIMENSION(:),POINTER,SAVE          :: TENTRY  ! entry time to JRCH (exit time u/s)
- INTEGER(I4B)                                :: NQ1     ! # flow particles
- ! (2) route flow within the current [JRCH] river segment
- INTEGER(I4B)                                :: ROFFSET ! retrospective offset due to rstep
- REAL(DP)                                    :: T_START ! start of time step
- REAL(DP)                                    :: T_END   ! end of time step
- REAL(DP),DIMENSION(:),POINTER,SAVE          :: T_EXIT  ! time particle expected exit JRCH
- LOGICAL(LGT),DIMENSION(:),POINTER,SAVE      :: FROUTE  ! routing flag .T. if particle exits
- INTEGER(I4B)                                :: NQ2     ! # flow particles (<=NQ1 b/c merge)
- ! (3) calculate time-step averages
- INTEGER(I4B)                                :: NR      ! # routed particles
- INTEGER(I4B)                                :: NN      ! # non-routed particles
- REAL(DP),DIMENSION(2)                       :: TNEW    ! start/end of time step
- REAL(DP),DIMENSION(1)                       :: QNEW    ! interpolated flow
- ! (4) housekeeping
- REAL(DP)                                    :: Q_END   ! flow at the end of the timestep
- REAL(DP)                                    :: TIMEI   ! entry time at the end of the timestep
- TYPE(FPOINT), DIMENSION(:), POINTER, SAVE   :: NEW_WAVE    ! temporary wave
- LOGICAL(LGT),SAVE                           :: INIT=.TRUE. ! used to initialize pointers
- ! random stuff
- INTEGER(I4B),PARAMETER                      :: MAXQPAR=20  ! maximum number of particles
- CHARACTER(LEN=256)                          :: CMESSAGE    ! error message for downwind routine
- ! initialize error control
- ierr=0; message='QROUTE_RCH/'
- !print*, 'jRch, ixOutlet = ', jRch, ixOutlet
- ! ----------------------------------------------------------------------------------------
- ! (0) INITIALIZE POINTERS
- ! ----------------------------------------------------------------------------------------
- IF(INIT) THEN
-  INIT=.FALSE.
-  NULLIFY(Q_JRCH,TENTRY,T_EXIT,FROUTE,NEW_WAVE)
- ENDIF
```

Comment:
I changed pointers (T_EXIT, Q_JRCH, TENTRY, FROUTE, NEW_WAVE) to allocatable array. See declarations above.   So I could have removed INITIALIZE POINTERS sections.